glad/parse.py

```diff
@@ -1453,3 +1445,90 @@ def from_element(cls, element, api=None):
     def __str__(self):
         return '{self.name}@{self.version!r}'.format(self=self)
     __repr__ = __str__
+
+
+class SpecificationDocs(object):
+    DOCS_NAME = None
+
+    SPEC = None
+    URL = None
+
+    def __init__(self, docs_dir):
+        self.docs_dir = docs_dir
+
+    def select(self, feature_set):
+        raise NotImplementedError
+
+    @classmethod
+    def default_out_dir(cls):
+        if cls.DOCS_NAME is None:
+            raise ValueError('DOCS_NAME not set')
+        return Path('.docs') / cls.DOCS_NAME
+
+    @classmethod
+    def from_url(cls, url, opener=None):
+        if opener is None:
+            opener = URLOpener.default()
+        docs_dir = cls.default_out_dir()
+        zip_out_dir = docs_dir.parent
+
+        with closing(opener.urlopen(url)) as f:
+            raw = f.read()
+            f = BytesIO(raw)
+
+            with zipfile.ZipFile(f) as zf:
+                zf.extractall(zip_out_dir)
+                zip_dir = zip_out_dir / zf.namelist()[0]
+                os.rename(zip_dir, docs_dir)
```

Comment:
I'd prefer if the docs worked with a zipfile without extracting it. Should be able to traverse the zip instead of the filesystem.

Reply:
Done :), didn't know it was possible